sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -68,4 +69,15 @@ default boolean canDeleteWhere(Filter[] filters) {
    * @throws IllegalArgumentException If the delete is rejected due to required effort
    */
   void deleteWhere(Filter[] filters);
+
+  Filter[] ALWAYS_TRUE_FILTER = new Filter[] { new AlwaysTrue() };
```

Comment:
This becomes a public API as well. Shall we put it in an internal object like `CatalogV2Util`?

Reply:
I wouldn't do that because of:
- I believe interfaces should be independent from internals as much as possible.
- ALWAYS_TRUE_FILTER can be used in other methods of the interface like `deleteWhere()`. For example, when an implementation overrides `truncateTable()` but an user want to delete all rows via `deleteWhere()`, so, he/she can re-use the constant.
- Other interfaces have constants too, for example https://github.com/apache/spark/blob/ec1560af251d2c3580f5bccfabc750f1c7af09df/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java#L47 . I don't see much difference between `PROP_LOCATION` and  `ALWAYS_TRUE_FILTER`.

Reply:
Can't people use `AlwaysTrue` directly if they want? I would also prefer to avoid having multiple ways to do the same thing. It looks odd to have the AlwaysTrue predicate constant as an API under `SupportsDelete` interface.

Reply:
How about to revert this commit https://github.com/apache/spark/pull/31475/commits/d1e5a18066f9fb2ff0ca1504e7c3f0802905febd , and implement it as:
```scala
default boolean truncateTable() {
    Filter[] filters = new Filter[] { new AlwaysTrue() };
   ...
```
I am not sure it is worth to do this premature optimization. Comparing to the truncation op, the allocation overhead is small. If it is a hot spot, JVM should do all work for us and optimize it, I do believe.

Reply:
Okay, I just noticed that it was changed per the review comment above. I think it's fine to remove as the overhead is small, and avoid exposing `ALWAYS_TRUE_FILTER` as an API.